internal/auth/auth.go

```diff
@@ -0,0 +1,106 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	// UserToken is the retrieved user token from the token file.
+	// It must be set using ReadToken.
+	UserToken string
+
+	// TokenDir is the directory for the token file.
+	TokenDir = ".config/benchttp" // nolint:gosec // no creds
+	// TokenName is the name for the token file.
+	TokenName = "token.txt"
+
+	// ErrTokenFind reports an error finding a token file.
+	ErrTokenFind = errors.New("token not found")
+	// ErrTokenRead reports an error reading a token file.
+	ErrTokenRead = errors.New("invalid token")
+	// ErrTokenSave reports an error saving a token file.
+	ErrTokenSave = errors.New("cannot save token")
+)
+
+// ReadToken reads a token from a file and sets UserToken to the retrieved
+// value or returns a non-nil error that is either ErrTokenFind or ErrTokenRead.
+func ReadToken() error {
```

Comment:
If i get it correctly, the flow is :

- setup a string var `UserToken`
- in `execute`, before exporting to server, load the file value inside `UserToken` via `ReadToken`.
- read non-nil `UserToken` inside `output/report` via `r.Header.Set("Authorization", "Bearer "+auth.UserToken)`

Seems convoluted and relying on implicit state.

Suggestion : 

```go
// cmd/benchttp/run.go

func execute() {
	//...
	token, err := auth.ReadToken()
	if err != nil {
		return err
	}
	err = output.New(rep, cfg).Export(token)
	if err != nil {
		return cmd.handleOutputError(err)
	}
	return nil
```

- `main` really passes all the args
- `output` package does not import `internal/auth`

Reply:
Was first implem, not that bad after all.
Note that it implies to store the token in `output` (either package scope or via a `Report.userToken`, because the token add to headers in method `Report.HTTP` that can't accept extra params because it implements `export.HTTPRequester` (adding it results in a silly signature). So in the end the flow is exactly the same: the only difference being _where_ the token is stored: `auth` vs `output`.